api/v1alpha1/componentversion_types.go

```diff
@@ -57,57 +58,70 @@ type ComponentVersionSpec struct {
 	ServiceAccountName string `json:"serviceAccountName,omitempty"`
 }
 
-// Repository defines the OCM Repository.
+// Repository specifies access details for the repository that contains OCM ComponentVersions.
 type Repository struct {
-	//TODO@souleb: do we need a scheme for the url?
-	// add description for each field
-	// Do we need a type field? (e.g. oci, git, s3, etc.)
+	// URL specifies the URL of an OCI repository in which the ComponentVersion is stored.
+	// +required
 	URL string `json:"url"`
 
+	// SecretRef specifies the credentials used to access the OCI registry.
```

Comment:
It needs to be documented somewhere that the key of the kubernetes secret and the name of the kubernetes secret have to be the same value

Reply:
The name of the kubernetes secret: https://github.com/open-component-model/ocm-controller/blob/main/e2e/testdata/testSignedOCIRegistryComponents/component_version.yaml#L19
has to be the same as the key in the kubernetes secret data key-value pair: https://github.com/open-component-model/ocm-e2e-framework/blob/main/shared/secrets.go#L37

The operator of the MPAS source cluster has to know this?

Reply:
You're referring to 2 different secrets @shivenduverma-sap. One of these holds the credentials to access an OCI repository. The other contains a secret used to verify signatures. This might be where you are getting confused as the key in the secret containing the public key needs to match the name of the signature. This is documented here: https://ocm.software/docs/controller-reference/ocm-controller/component-version/#public-key